src/main/java/com/lucidworks/spark/SchemaPreservingSolrRDD.java

```diff
@@ -0,0 +1,424 @@
+package com.lucidworks.spark;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.log4j.Logger;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.common.SolrDocument;
+import org.apache.solr.common.SolrDocumentList;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.api.java.function.Function;
+import org.apache.spark.mllib.linalg.*;
+import org.apache.spark.sql.RowFactory;
+
+import org.apache.spark.sql.types.*;
+import org.apache.spark.sql.Row;
+
+/**
+ * Created by ganeshkumar on 9/29/15.
```

Comment:
Please provide a useful description for this class or remove the JavaDoc comment